man/adoc/bpftrace.adoc

```diff
@@ -466,6 +467,33 @@ If you need another representation, you may cast to the following built in types
 
 |int64
 |Signed 64 bit integer
+
+|buf_t[]
+|Buffer of bytes. Returned by <<functions-buf, `buf`>>
+
+|cgroup_path_t
+|Formatted cgroup path. Returned by <<functions-cgroup_path, `cgroup_path`>>
+
+|inet_t[]
+|IPv4 or IPv6 address. Returned by <<functions-ntop, `ntop`>>
+
+|ksym_t
+|Kernel symbol. Returned by <<functions-ksym, `ksym`>>
+
+|macaddr_t
+|Macaddress. Returned by <<functions-macaddr, `macaddr`>>
+
+|strerror_t
+|String represenation of errno code. Returned by <<functions-strerror, `strerror`>>
+
+|str_t[]
+|String: Null terminated array of characters
```

Comment:
I'm a bit wary of documenting these builtin types as I'm not sure that they're implemented correctly right now. Strings, for example, shouldn't be an array type - they're variable length - and I'm not sure that we should specify that they're null terminated as that's an implementation detail that we might want to change.

Also do `inet_t` and `buf_t` need to be array types?

Reply:
I agree but we should document them somehow before we land this PR.

Another point - all all built-in types usable in a cast? If not, should we mention that in the docs?

It's starting to look like we may need a separate PR to clarify types documentation :grinning:

Reply:
> Also do inet_t and buf_t need to be array types?

They do.

> all built-in types usable in a cast?

No. All the aggregate map types like `sum_t` and `max_t` are not allowed. However, it seems like I need to add an additional semantic_analyser check for those. So good question!

> It's starting to look like we may need a separate PR to clarify types documentation 😀

I feel like it doesn't make sense as a PR **before** this one - so maybe we just leave the adoc alone for now. I'll remove the additional types I added from the Data Types section. Then I'll put up a separate PR where we can discuss how we want to handle type documentation.